### PR TITLE
Filemanager copy/move: add support of centralized sdr folders

### PR DESCRIPTION
**Pull request by hius07**
`pasteHere()` fixed and optimized.

Support for centralized sdr storage provided by the patches https://github.com/koreader/koreader/issues/4831#issuecomment-548245804 (a file with a doc metadata in koreader/history) and https://github.com/koreader/koreader/issues/9265#issuecomment-1382729422 (a folder with doc metadata in a user defined folder).

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10051)

### CODE REVIEW

**Comment by poire-z** [comment]
Hard to follow the diff (even if small), so trusting you on the changes.

But.... what are "centralized sdr folders" ?
Do you mean the fallback koreader/history/ that is used on Android when the filesystem where the book lies is readonly ? Or something else ?
And would it work with the user patch provided at https://github.com/koreader/koreader/issues/9265#issuecomment-1382729422 (that I haven't tried, but if somehow your changes make our features 100% working with that user patch, that's pretty nice).

**Comment by hius07** [comment]
Yes, for that patch.
I do not see other changes reqiured for 100% working of the centralized sdr storage.
Going to build it in, switching from UI.
(Desperated users build Android packages with centralized sdr storage, they are not aware of user patches: https://4pda.to/forum/index.php?showtopic=461675&st=285200#entry120350005))

**Comment by poire-z** [comment]
(Your link gives me a 404.)
Good, you're probably the right and best person to implement #9265 :)
Give some thoughts to my various warnings in that thread, ie. https://github.com/koreader/koreader/issues/9265#issuecomment-1171444710 (ie. don't forget reading/migrating old sidecar in koreader/history).

**Comment by hius07** [comment]
The first post updated.

**Comment by hius07** [comment]
I see we make similar sidecar checks/updates for rename/delete/copy/move file operations.
Maybe add a method `DocSettings:updateSidecar(old_file, new_file, move)`?

**Comment by hius07** [comment]
I wonder which place is better: sidecars are hadled by DocSettings, but updates of sidecar files are requested by Filemanager only.

**Comment by poire-z** [comment]
Depends on the level of changes or refactoring you allow yourself :)
If the main object the other code should handle is a DocSettings object, and DocSettings object have various ways to persist on the filesystem (some kind with a sidecar directory, some other kind without any sidecar directory, like those in koreader/history/), the internal handling of such sidecar directory is better left to the object itself, and should be hidden from the caller. That is, ideally, FileManager shouldn't have to have different bits of code to handle the case whether it has or not a sidecar directory to `cp -r`, or just a file to rename (or tomorrow, whether it's saved in a DB or remotely in some cloud storage :)
So, ideally, this being handled by DocSettings itself feels better to me.

But it might be simpler and more practical and less kludgy (so, better :) to have it handled by FileManager.
Up to you again :)

File: frontend/apps/filemanager/filemanager.lua [hunk 1]
```diff
@@ -989,12 +989,19 @@ function FileManager:pasteHere(file)
     local function updateSidecar(move)
         if DocSettings:hasSidecarFile(orig) then
             local dest_sidecar_dir = DocSettings:getSidecarDir(dest_file)
-            BaseUtil.execute(self.mkdir_bin, dest_sidecar_dir)
+            DocSettings:ensureSideca(dest_sidecar_dir)
```
Comment (NiLuJe) [inline frontend/apps/filemanager/filemanager.lua:992]: typo? ensureSideca***r***

**Comment by hius07** [comment]
Closing in favor of more generic solution.
